src/Stratis.Bitcoin.Tests/BlockPulling2/BlockPullerBehaviorTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using Moq;
+using Stratis.Bitcoin.BlockPulling2;
+using Stratis.Bitcoin.Configuration.Logging;
+using Stratis.Bitcoin.Interfaces;
+using Xunit;
+
+namespace Stratis.Bitcoin.Tests.BlockPulling2
+{
+    public class BlockPullerBehaviorTests
+    {
+        private readonly BlockPullerBehavior behavior;
+
+        public BlockPullerBehaviorTests()
+        {
+            var puller = new Mock<IBlockPuller>();
+
+            var ibdState = new Mock<IInitialBlockDownloadState>();
+            ibdState.Setup(x => x.IsInitialBlockDownload()).Returns(() => true);
+
+            var loggerFactory = new ExtendedLoggerFactory();
+            loggerFactory.AddConsoleWithFilters();
+
+            this.behavior = new BlockPullerBehavior(puller.Object, ibdState.Object, loggerFactory);
+        }
+
+        [Fact]
+        public void InitializesWithDefaultValues()
+        {
+            Assert.Equal(BlockPullerBehavior.SamplelessQualityScore, this.behavior.QualityScore);
+            Assert.Equal(0, this.behavior.SpeedBytesPerSecond);
+            Assert.Null(this.behavior.Tip);
+        }
+
+        [Fact]
+        public void QualityScoreCanGoToMinimum()
+        {
+            this.behavior.Penalize(10, 10);
+            this.behavior.RecalculateQualityScore(1000);
+
+            Assert.Equal(BlockPullerBehavior.MinQualityScore, this.behavior.QualityScore);
```

Comment:
Is it not double?

Reply:
It is, but since no math operations on this number were performed it should be fine. 

So smth like 
```
double a = 5.0;
bool equal = a == 5.0
```
Should always execute. 

But smth like 

```
double a = (5.0 / 10) * 10;
bool equal = a == 5.0
```

will not always be true


In this test I'm testing that the last operation on quality score was qualityscore = minqualityScore

Reply:
Probably okay